src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -94,7 +94,8 @@ internal static RuntimeGraph GetRuntimeGraph(IEnumerable targetFrameworks)
         {
             var tfi = new TargetFrameworkInformation
             {
-                FrameworkName = NuGetFramework.Parse(targetFrameworkInfo.TargetFrameworkMoniker)
+                FrameworkName = GetTargetFramework(targetFrameworkInfo.Properties),
+                TargetAlias = GetPropertyValueOrNull(targetFrameworkInfo.Properties, ProjectBuildProperties.TargetFramework)
```

Comment:
Populate the alias in ProjectReference metadata info. 
Otherwise, no-op will fail consistently :)